README.md

```diff
@@ -1 +1,85 @@
-# gcp-workload-identity-federation-buildkite-plugin
\ No newline at end of file
+# Google Cloud Workload Identity Federation Buildkite Plugin
+
+A Buildkite plugin to assume a Google Cloud service account using [workload identity federation](https://cloud.google.com/iam/docs/workload-identity-federation).
+
+The plugin requests an OIDC token from Buildkite and uses it to a populate Google Cloud credentials file.
+
+The path to the file is populated in `GOOGLE_APPLICATION_CREDENTIALS` for SDKs that use [Application Default Credentials](https://cloud.google.com/docs/authentication/application-default-credentials), and in `CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE` for the `gcloud` CLI.
+
+## Google Cloud configuration
+
+You should already have a Google Cloud project and a Service Account to assume. See [Google's documentation](https://cloud.google.com/iam/docs/workload-identity-federation-with-other-providers) for more detailed instructions for these steps.
+
+1. Create a [Workload Identity Pool](https://console.cloud.google.com/iam-admin/workload-identity-pools).
+
+   We recommend creating a different pool for each security boundary.
+
+   In this example we're using `buildkite-example-pipeline`.
+
+2. Add a provider to the pool.
+
+   Use OpenID Connect, and give it a name like `buildkite`.
+
+   Use `https://agent.buildkite.com` as the Issuer.
+
+   Copy the value of the default audience or provide your own.
+
+3. Configure provider attributes.
+
+   We suggest the following mapping:
+
+   | Google | OIDC |
+   | --- | --- |
+   | google.subject | assertion.sub |
+   | attribute.pipeline_slug | assertion.pipeline_slug |
+   | attribute.build_branch | assertion.build_branch |
```

Comment:
I recall @pda having some 🧠 around which buildkite-supplied claims are best to use for having a robust security boundary for IAM kinda things